File: examples/framework-adapters/openai.py
from runx import RunxClient, create_framework_bridge, create_openai_adapter


def main() -> None:
    client = RunxClient()
    bridge = create_framework_bridge(client)
    adapter = create_openai_adapter(bridge)
    response = adapter.run(
        "skills/sourcey",
        inputs={"project": "."},
        resolver=lambda context: True if context.request.get("kind") == "approval" else None,
    )
    print(response)


if __name__ == "__main__":
    main()

File: examples/framework-adapters/openai.ts
import { createFrameworkBridge, createOpenAiAdapter, createRunxSdk } from "@runx/sdk";

async function main(): Promise<void> {
  const sdk = createRunxSdk({ callerOptions: { maxAttempts: 1 } });
  const bridge = createFrameworkBridge({ execute: sdk.runSkill.bind(sdk) });
  const openai = createOpenAiAdapter(bridge);

  const response = await openai.run({
    skillPath: "skills/sourcey",
    inputs: { project: "." },
    resolver: ({ request }) => {
      if (request.kind === "approval") {
        return true;
      }
      return undefined;
    },
  });

  console.log(JSON.stringify(response, null, 2));
}

void main();
